### DISCUSSION TITLE
config item to follow redirects by default

### DISCUSSION THREAD

**Discussion by AntiSol** [encode/httpx#2598] [Ideas] [answered]
Hello.

I wasn't sure if I should reopen/comment on #2126 or make a new issue.

I've recently run into this change to the default behaviour around following redirects, and I'd like to be able to easily refactor my large codebase to set the default behaviour back. I'd also like be able to change the default to prevent other coders who may be unaware of this behaviour from inadvertently doing a `get(url)` without specifying `follow_redirects` (we almost always want to follow redirects in our codebase)

What I'd like to do is have a new config item in httpx._config similar to `DEFAULT_MAX_REDIRECTS`. This would be called something like `DEFAULT_FOLLOW_REDIRECTS` and would determine the default behaviour of `follow_redirects` is not provided

I'd like to do this to allow for usage like:
```python
import httpx
httpx.DEFAULT_FOLLOW_REDIRECTS = True
httpx.get(some_url)
```

rather than having to either pass `follow_redirects` as a param to get or instantiate and persist a `Client` with `follow_redirects=True`

I think I could put together a PR to do this change, would that be likely to be accepted if I were to do so? Is there anything I should know before I write the code? :) 

Thanks

**Comment by lovelydinosaur** [comment, marked as answer]
Thank you for the suggestion, appreciate your involvement.

I don't think we'd be likely to accept a package-level config item. It wouldn't be consistent with the API that we're using elsewhere throughout this project.

**Comment by AntiSol** [comment]
maybe I'm missing something. How would a DEFAULT_FOLLOW_REDIRECTS be inconsistent with the existing DEFAULT_MAX_REDIRECTS?

**Reply by lovelydinosaur** [reply]
`DEFAULT_MAX_REDIRECTS` isn't public API. It's a bit of private implementation detail.
